io-sim/src/Control/Monad/IOSimPOR/Internal.hs

```diff
@@ -570,8 +570,9 @@ schedule thread@Thread{
       written <- execAtomically' (runSTM $ writeTVar tvar TimeoutCancelled)
       (wakeup, wokeby) <- threadsUnblockedByWrites written
       mapM_ (\(SomeTVar var) -> unblockAllThreadsFromTVar var) written
+      written' <- mapM someTVarToLabelled written
```

Comment:
could you perhaps move this above to line 570?

Reply:
Done